articles/active-directory/develop/msal-net-client-assertions.md

```diff
@@ -116,30 +123,30 @@ string Encode(byte[] arg)
     return s;
 }
 
-string GetSignedClientAssertion()
+static string GetSignedClientAssertion(string tenantId, string clientId)
 {
-    //Signing with SHA-256
-    string rsaSha256Signature = "http://www.w3.org/2001/04/xmldsig-more#rsa-sha256";
-     X509Certificate2 certificate = new X509Certificate2("Certificate.pfx", "Password", X509KeyStorageFlags.EphemeralKeySet);
+    // It's not recommended to hardcode a password or a file location. Get the X509Certificate2 from somewhere save.
+    X509Certificate2 certificate = new X509Certificate2("Certificate.pfx", "Password", X509KeyStorageFlags.EphemeralKeySet);
 
-    //Create RSACryptoServiceProvider
-    var x509Key = new X509AsymmetricSecurityKey(certificate);
-    var privateKeyXmlParams = certificate.PrivateKey.ToXmlString(true);
-    var rsa = new RSACryptoServiceProvider();
-    rsa.FromXmlString(privateKeyXmlParams);
+    // Get the RSA with the private key, used for signing.
+    var rsa = certificate.GetRSAPrivateKey();
```

Comment:
We use `RSACng rsa = certificate.GetRSAPrivateKey() as RSACng;` internally

Reply:
> The RSACng class is an implementation of the RSA algorithm using the Windows CNG libraries and isn't available on operating systems other than Windows. For applications which aren't doing Windows interop, you're encouraged to use RSA.Create instead of referencing this type directly.
> 
> https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.rsacng?view=net-6.0

They suggest to **not** reference it this way.

Reply:
@bgavrilMS are you ok with leaving out the `as RSACng` part?

Reply:
Interesting, as we've got a lot of ppl running this code on Linux backends :)

Reply:
But sure, let's go ahead with `var rsa = certificate.GetRSAPrivateKey();`